utils.sh

```diff
@@ -99,6 +99,21 @@ function create_cluster() {
 
     trap auth_template_and_removetmp EXIT
     $OPENSHIFT_INSTALLER --dir "${assets_dir}" --log-level=debug create cluster
+
+    # Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1816904
+    wait_for_workers
+}
+
+wait_for_workers() {
+    while (( $(oc get nodes | grep "^worker" | wc -l) < $NUM_WORKERS )); do
+        echo "Waiting for $NUM_WORKERS workers"
+        sleep 5
+    done
```

Comment:
What if, for some reason, one of the node does not come up? Maybe a timeout could be useful also here (even though in the CI we have an outer timeout wrapping the dev-script setup)?

Reply:
That's a good point - with the multi-stage jobs will we still do the must-gather if this loops forever and the job times out?

Reply:
Yes. We gather the logs in case of workflow failure, and the multistage step fails due the timeout

Reply:
Can we limit the amount of time this runs for? I'm concerned now that CI jobs where we have a worker failure will consume the entire 4 hours.

Reply:
I think that the [current setup timeout]([url](https://github.com/openshift/release/blob/447728c06eb71cb7ea80f40899c51c77fa7e12c5/ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh#L109)) will protect against that. In any case a limit could be useful in general, also in other execution environment (ie local deploy)

Reply:
Ok sounds good, I'll update to add a timeout :+1: